frontend/src/plugin/registry.tsx

```diff
@@ -559,4 +560,11 @@ export function registerGetTokenFunction(override: (cluster: string) => string |
   store.dispatch(setFunctionsToOverride({ getToken: override }));
 }
 
+export function registerPluginSettings(
+  name: string,
+  component: PluginSettingsComponentType,
+  autoSave: boolean = false
```

Comment:
I'd make these args as options, so we can easily extend it in the future.
Also, the autoSave is not very explicit as a name. I would actually guess the opposite of what it does.
So calling it `withSaveButton: boolean = true` is more explicit IMO (and we need to explain anyway that for the save button to take effect, the component needs to use the onDataChanged).

Reply:
have changed it to `displaySaveButton`.